pkg/pillar/cmd/zedmanager/handlevolumemgr.go

```diff
@@ -281,44 +302,70 @@ func handleVolumesSnapshotStatusModify(ctx interface{}, key string, status inter
 		log.Errorf("Snapshot handling %s failed: %s", volumesSnapshotStatus.SnapshotID, volumesSnapshotStatus.Error)
 		appInstanceStatus.SetErrorWithSourceAndDescription(volumesSnapshotStatus.ErrorDescription, volumesSnapshotStatus.ErrorSourceType)
 		setSnapshotStatusError(appInstanceStatus, volumesSnapshotStatus.SnapshotID, volumesSnapshotStatus.ErrorDescription)
-		publishAppInstanceStatus(zedmanagerCtx, appInstanceStatus)
+		publishAppInstanceStatus(ctx, appInstanceStatus)
 		return
 	}
 	appInstanceStatus.SnapStatus.RollbackInProgress = false
-	publishAppInstanceStatus(zedmanagerCtx, appInstanceStatus)
+	// unpublish local app instance config, as the rollback is complete, we do not need to replace the config
+	config := lookupAppInstanceConfig(ctx, appInstanceStatus.Key(), false)
+	if config == nil {
+		log.Fatalf("reactToSnapshotRollback: AppInstanceConfig not found for %s", appInstanceStatus.Key())
+	}
+	unpublishLocalAppInstanceConfig(ctx, appInstanceStatus.Key())
+	publishAppInstanceStatus(ctx, appInstanceStatus)
+	doUpdate(ctx, *config, appInstanceStatus)
 }
 
-func handleVolumesSnapshotStatusDelete(ctx interface{}, key string, status interface{}) {
-	log.Noticef("handleVolumesSnapshotStatusDelete")
-	volumesSnapshotStatus := status.(types.VolumesSnapshotStatus)
-	zedmanagerCtx := ctx.(*zedmanagerContext)
-	appInstanceStatus := lookupAppInstanceStatus(zedmanagerCtx, volumesSnapshotStatus.AppUUID.String())
+func reactToSnapshotCreate(ctx *zedmanagerContext, volumesSnapshotStatus types.VolumesSnapshotStatus) {
+	log.Noticef("reactToSnapshotCreate")
+	appInstanceStatus := lookupAppInstanceStatus(ctx, volumesSnapshotStatus.AppUUID.String())
 	if appInstanceStatus == nil {
-		log.Errorf("handleVolumesSnapshotStatusDelete: AppInstanceStatus not found for %s", volumesSnapshotStatus.AppUUID.String())
+		log.Errorf("reactToSnapshotCreate: AppInstanceStatus not found for %s", volumesSnapshotStatus.AppUUID.String())
 		return
 	}
-	if volumesSnapshotStatus.HasError() {
+
+	if volumesSnapshotStatus.HasError() && volumesSnapshotStatus.ErrorSeverity != types.ErrorSeverityWarning {
```

Comment:
Why don't you propagate the warnings? Is due to the warning you generate in createSnapshot when the snapshot already exists? If so, please add a comment about this.